### PR TITLE
Make rollkit ibc compatible

### PR DESCRIPTION
**Pull request by catShaark**
## Overview

ref: #1319

The goal here is to make rollkit ibc compatible.

### Changes:
- We temporarily add `ValidatorHash` to rollkit header so that the `ValidatorHash` can be passed into the corresponding abci header. We're finding a way to remove `ValidatorHash` and still make it work.
- Fix rollkit not updating app hash per block
- Change in commit logic: senquencer will sign the `consensusVote` instead of header. This needed for light client compatible, specifically the signature verify in [`verifyCommitBatch`](https://github.com/cometbft/cometbft/blob/12713f6227a10b1f4531710b5a42c074b8ca50ca/types/validation.go#L153)
- Make change when convert rollkit commit to ABCI commit, we still need to add `timestamp` and `val address` fields in ABCI commit for light client compatible : https://github.com/vuong177/rollkit/blob/dd6b6ac7bd8769f50b2a40470708129989069d57/types/block.go#L76

## Checklist

- [ ] New and updated code has appropriate documentation
- [ ] New and updated code has new and/or updated testing
- [ ] Required CI checks are passing
- [ ] Visual proof for any user facing features like CLI or documentation updates
- [ ] Linked issues closed with keywords

### CODE REVIEW

**Comment by S1nus** [comment]
thanks, looks great!

I actually am [trying to remove](https://github.com/rollkit/rollkit/pull/1368/files#diff-fef4d0c951abc3c3c62f09cb5f74299506b4bca6d17c830841edced5e1971bf0L19) `SignedHeader.Validators` right now, to make our centralized sequencer code easier to reason about.

After we make sure it works, we can carefully add it back to enable IBC compatibility.

**Comment by catShaark** [comment]
I think there could be a way to remove the field and still be IBC-compatible. Should we proceed that direction ?

File: node/full_client.go [hunk 1]
```diff
@@ -486,7 +486,11 @@ func (c *FullClient) Commit(ctx context.Context, height *int64) (*ctypes.ResultC
 	if err != nil {
 		return nil, err
 	}
-	commit := com.ToABCICommit(heightValue, b.Hash())
+
+	// we only have one centralize sequencer = 1 validators
+	val := b.SignedHeader.Validators.Validators[0].Address
+	commit := com.ToABCICommit(heightValue, b.Hash(), val, b.SignedHeader.Time())
```
Comment (S1nus) [inline node/full_client.go:492]: Rollkit doesn't have validators. At the moment, we want to get the centralized sequencer from the genesis, and keep track of their pubkey in `Header.ProposerPubkey`.

Validators will be carefully added back for IBC compatibility in the near future

Comment (S1nus) [reply node/full_client.go:492]: not so sure yet. can we discuss after https://github.com/rollkit/rollkit/pull/1368 is merged?

Comment (S1nus) [reply node/full_client.go:492]: Ok nevermind, we're keeping the Validators field.
We might merge this!

Comment (vuong177) [reply node/full_client.go:492]: @S1nus yes, easier to keep rollup IBC compatible when we have valset info in rollkit header

**Comment by catShaark** [comment]
close this PR for now, will open a new one

**Pull request closed without merging**